tasks/mod-extractor.js

```diff
@@ -10,62 +10,48 @@ governing permissions and limitations under the License.
 */
 
 const fs = require("fs");
+const fsp = fs.promises;
 const path = require("path");
-const readline = require("readline");
+const fg = require("fast-glob");
 
-const pattern = "--mod-";
-const rootDir = path.join(__dirname, "..", "components");
-const fileExtension = ".css";
-const writePath = "metadata/mods.md";
-
-function extract(line, index) {
-    let modVariable = "";
-    for(let i = index; i < line.length; i++) {
-        if (line[i] == ")" || line[i] == "," || line[i] == " ") {
-            break;
+async function main(dir = path.join(__dirname, "../components")) {
+    for(const filepath of await fg('*/*.css', {
+        cwd: dir,
+        absolute: true,
+    })) {
+        if (filepath.includes('vars') || filepath.includes('tokens')) {
+            continue;
         }
-        modVariable += line[i];
-    }
-    return modVariable;
-}
 
-function writeToFile(file) {
-    const directoryName = path.dirname(file);
-    const readInterface = readline.createInterface({
-        input: fs.createReadStream(file),
-        console:false
-    });
-    var result  = new Set();
-    readInterface.on("line", function(line) {
-        let index = line.indexOf(pattern);
-        while(index != -1) {
-            let modVariable = extract(line, index);
-            result.add(modVariable);
-            index = line.indexOf(pattern, index + 1);
-        }
-        if (result.size) {
-            const resultArray = [...result];
-            const formattedResults = resultArray.map((result) => `|\`${result}\`|`);
-            const tableHeader = `| Modifiable Custom Properties |\n| --- |`;
-            formattedResults.unshift(tableHeader);
-            fs.writeFile(`${directoryName}/${writePath}`, formattedResults.join('\n'), (err) => {});
-        }
-    });
-}
+        const regex = /var\((--mod-(?:\w|-)+)/g;
+        const matches = await fsp.readFile(filepath, 'utf-8').then((content) => {
+            return [...content.matchAll(regex)].map((match) => match[1]);
+        }).catch((err) => {
+            console.log(err);
+        });
+
+        if (!matches || matches.length === 0) continue;
+
+        const found = [...new Set(matches)].sort();
```

Comment:
🤩

Reply:
Great!! A small suggestion. It would be great if you can add some code comments too?!!!

Reply:
Just pushed up some more comments! Thanks for the suggestion!